pkg/gateway/sig/javav2.go

```diff
@@ -0,0 +1,146 @@
+package sig
+
+import (
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"net/http"
+	"net/url"
+	"sort"
+	"strings"
+
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	"github.com/treeverse/lakefs/pkg/gateway/errors"
+	"github.com/treeverse/lakefs/pkg/logging"
+)
+
+// Implements the "signing protocol" as exists at aws-sdk-java @ 1.12.390 (commit 07926f08a70).
+// This should never have worked, on any version of S3, but the implementation is there, unmodified since 2015.
+// The code is below, for reference
```

Comment:
I am not sure why we need this.  The Java client receives some presigned URL and uses it.  AFAIU Java never signs anything in this version.  How can lakeFSFS even _know_ which version encoded the URL that it received?

Reply:
I think you are mixing S3A with lakeFSFS.
In our case setting the signing protocol to `QueryStringSignerType` ensures us that the redirect request will not be re-signed before being sent

Reply:
Ouch!

But still: we _need_ to provide a broken incorrect sigV2 signer?  Because we already have really good sigV4 and sigV2 signers!